src/NServiceBus.Core/Transports/TransportDefinition.cs

```diff
@@ -43,7 +43,7 @@ public abstract partial class TransportDefinition
         /// <summary>
         /// Returns the discriminator for this endpoint instance.
         /// </summary>
-        public abstract string GetDiscriminatorForThisEndpointInstance(ReadOnlySettings settings);
+        public abstract EndpointInstance BindToLocalEndpoint(EndpointInstance instance, ReadOnlySettings settings);
```

Comment:
I don't think it conveys the same intent which is "return an endpoint instance which is specific for this local endpoint", hence "bind".